src/common/telemetry/src/metric.rs

```diff
@@ -60,24 +60,47 @@ pub fn try_handle() -> Option<PrometheusHandle> {
 pub struct Timer {
     start: Instant,
     name: &'static str,
+    labels: Option<Vec<(String, String)>>,
```

Comment:
Just use `Vec<(String, String)>` here so we can update the vec with `get_labels` below. Otherwise if labels is `None`, there is no chance to change even if we borrow it as mutable.